src/main/java/uk/gov/mca/beacons/service/serializer/BeaconsSearchResultSerializer.java

```diff
@@ -0,0 +1,180 @@
+    } catch (IOException e) {
+      throw new BeaconsSearchResultSerializerException(
+        "failed to serialize BeaconSearchResult (IOException)",
+        e
+      );
+    }
+  }
+
+  private void writeData(BeaconsSearchResult value, JsonGenerator gen)
+    throws IOException {
+    gen.writeFieldName("data");
+    gen.writeStartArray();
+    value
+      .getBeacons()
+      .forEach(
+        beacon -> {
+          try {
+            gen.writeStartObject();
+            writeBeaconDetails(gen, beacon);
+            gen.writeEndObject();
+          } catch (IOException e) {
+            throw new BeaconsSearchResultSerializerException(
+              "failed to serialize beacon data for beaconId:" + beacon.getId(),
+              e
+            );
+          }
+        }
+      );
+    gen.writeEndArray();
+  }
+
+  private void writeMeta(BeaconsSearchResult value, JsonGenerator gen)
+    throws IOException {
+    gen.writeObjectFieldStart("meta");
+    gen.writeNumberField("count", value.getCount());
+    gen.writeNumberField("pageSize", value.getPageSize());
+    gen.writeEndObject();
+  }
+
+  private void writeEmergencyContacts(JsonGenerator gen, Beacon beacon)
+    throws IOException {
+    gen.writeFieldName("emergencyContacts");
+    gen.writeStartArray();
+    if (beacon.getEmergencyContacts() != null) {
+      beacon
+        .getEmergencyContacts()
+        .forEach(
+          contact -> {
+            try {
+              gen.writeStartObject();
+              gen.writeStringField("fullName", contact.getFullName());
+              gen.writeStringField(
+                "telephoneNumber",
+                contact.getTelephoneNumber()
+              );
+              gen.writeStringField(
+                "alternativeTelephoneNumber",
+                contact.getAlternativeTelephoneNumber()
+              );
+              gen.writeEndObject();
+            } catch (IOException e) {
+              throw new BeaconsSearchResultSerializerException(
+                "failed to serialize emergency contacts for beaconId:" +
+                beacon.getId(),
+                e
+              );
+            }
+          }
+        );
+    }
+
+    gen.writeEndArray();
+  }
+
+  private void writeBeaconDetails(JsonGenerator gen, Beacon beacon)
+    throws IOException {
+    gen.writeStringField("type", "beacon");
+    gen.writeStringField("id", beacon.getId().toString());
+    gen.writeObjectFieldStart("attributes");
+
+    gen.writeStringField("manufacturer", beacon.getManufacturer());
+    gen.writeStringField("model", beacon.getModel());
+    gen.writeStringField(
+      "manufacturerSerialNumber",
+      beacon.getManufacturerSerialNumber()
+    );
+    gen.writeStringField("chkCode", beacon.getChkCode());
+    gen.writeStringField(
+      "batteryExpiryDate",
+      beacon.getBatteryExpiryDate().toString()
+    );
+    gen.writeStringField(
+      "lastServicedDate",
+      beacon.getLastServicedDate().toString()
```

Comment:
These fields can be null as the user isn't required to submit a battery expiry or last serviced date.

Reply:
wrapped the call to writeStringField(...)